<launch>
  <!-- Namespace -->
  <arg name="robot_frame" default="base_footprint"/>
  <arg name="utm_frame" default="utm"/>
  <arg name="odom_frame" default="odom"/>
  <arg name="cam_frame" default="raspicam"/>

  <node pkg="waypoint_nav" type="fiducial_waypoint_localization.py" name="fiducial_waypoint_localization">
    <param name="waypoint_control/base_frame" value="$(arg robot_frame)"/>
    <param name="waypoint_control/gps_frame" value="$(arg utm_frame)" />
    <param name="waypoint_control/odom_frame" value="$(arg odom_frame)"/>
    <param name="waypoint_control/map_file" value="$(find waypoint_nav)/src/Fiducials.json"/>
    <param name="waypoint_control/camera_frame" value="$(arg cam_frame)"/>
    <remap from="robot_gps_pose" to="odo_calib_pose"/>
  </node>  
  <node pkg="waypoint_nav" type="encoder_waypoint_localization.py" name="encoder_waypoint_localization">
    <param name="waypoint_control/base_frame" value="$(arg robot_frame)"/>
    <param name="waypoint_control/gps_frame" value="$(arg utm_frame)" />
    <param name="waypoint_control/odom_frame" value="$(arg odom_frame)"/>
  </node>
  <node pkg="waypoint_nav" type="3d_waypoint_control.py" name="waypoint_control" >
    <param name="robot_x_config" value="True"/>
    <param name="robot_y_config" value="False"/>
    <param name="robot_z_config" value="False"/>
    <param name="robot_rx_config" value="False"/>
    <param name="robot_ry_config" value="False"/>
    <param name="robot_rz_config" value="True"/>
  </node>
  <node pkg="waypoint_nav" type="waypoint_prediction.py" name="waypoint_prediction"/>
</launch>
